paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentSelection.kt

```diff
@@ -395,3 +396,35 @@ internal val PaymentSelection.paymentMethodType: String
         is PaymentSelection.New.USBankAccount -> paymentMethodCreateParams.typeCode
         is PaymentSelection.Saved -> paymentMethod.type?.name ?: "card"
     }
+
+internal val PaymentSelection.billingDetails: PaymentMethod.BillingDetails?
+    get() = when (this) {
+        is PaymentSelection.ExternalPaymentMethod -> billingDetails
+        PaymentSelection.GooglePay -> null
+        PaymentSelection.Link -> null
+        is PaymentSelection.New.Card -> paymentMethodCreateParams.billingDetails
+        is PaymentSelection.New.GenericPaymentMethod -> paymentMethodCreateParams.billingDetails
+        is PaymentSelection.New.LinkInline -> linkPaymentDetails.paymentMethodCreateParams.billingDetails
+        is PaymentSelection.New.USBankAccount -> paymentMethodCreateParams.billingDetails
+        is PaymentSelection.Saved -> paymentMethod.billingDetails
+    }
+
+internal fun PaymentMethod.BillingDetails?.toPaymentSheetBillingDetails(): PaymentSheet.BillingDetails? {
+    return if (this == null) {
```

Comment:
Seems like we should just accept a non null param here, and do optional chaining (aka foobar?.toPaymentSheetBillingDetails())